### ISSUE TITLE
How to add new learnable parameters to the optimizer?

### ISSUE THREAD

**Issue by jodumagpi** [open-mmlab/mmdetection#4967]
If I want to add a learnable parameter to the loss function, say a weighing parameter, how can it be added to the list of parameters that the optimizer updates?

I just want to add that I know how to modify the codes for the loss function and how to add it to the config. In fact, I had successfully trained my own dataset with a modified loss function. However, I do not know how to add a new learnable parameter that I defined in the loss function (using torch.nn.Parameter) to the optimizer after it has already been initialized/built. Is there any way to do this?

**Comment by jodumagpi** [comment]
I just want to add that I know how to modify the codes for the loss function and how to add it to the config. In fact, I had successfully trained my own dataset with a modified loss function. However, I do not know how to add a new learnable parameter that I defined in the loss function (using torch.nn.Parameter) to the optimizer after it has already been initialized/built. Is there any way to do this?

**Comment by jodumagpi** [comment]
Nevermind. I just realized that the loss functions are already included as part of the model. Then, whatever learnable parameters that I declare inside the model will be automatically added 
to the parameter list that the optimizer updates.
